crates/router/src/core/payment_methods/network_tokenization.rs

```diff
@@ -0,0 +1,700 @@
+}
+
+#[derive(Debug, Deserialize, Eq, PartialEq)]
+pub struct NetworkTokenErrorResponse {
+    error_message: String,
+    error_info: NetworkTokenErrorInfo,
+}
+
+#[derive(Debug, Deserialize, Eq, PartialEq)]
+pub struct DeleteNetworkTokenResponse {
+    status: DeleteNetworkTokenStatus,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CheckTokenStatus {
+    card_reference: String,
+    customer_id: id_type::CustomerId,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum TokenStatus {
+    Active,
+    Inactive,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CheckTokenStatusResponsePayload {
+    token_expiry_month: Secret<String>,
+    token_expiry_year: Secret<String>,
+    token_status: TokenStatus,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct CheckTokenStatusResponse {
+    payload: CheckTokenStatusResponsePayload,
+}
+
+pub async fn mk_tokenization_req(
+    state: &routes::SessionState,
+    payload_bytes: &[u8],
+    amount: String,
+    currency: String,
+    customer_id: id_type::CustomerId,
+) -> CustomResult<(CardNetworkTokenResponsePayload, Option<String>), errors::NetworkTokenizationError>
+{
+    let tokenization_service = &state.conf.network_tokenization_service.get_inner();
+    let enc_key = tokenization_service.public_key.peek().clone();
+
+    let key_id = tokenization_service.key_id.clone();
+
+    let jwt = encryption::encrypt_jwe(
+        payload_bytes,
+        enc_key,
+        EncryptionAlgorithm::A128GCM,
+        Some(key_id.as_str()),
+    )
+    .await
+    .change_context(errors::NetworkTokenizationError::SaveNetworkTokenFailed)
+    .attach_printable("Error on jwe encrypt")?;
+
+    let order_data = OrderData {
+        consent_id: uuid::Uuid::new_v4().to_string(),
+        customer_id,
+        amount,
+        currency,
+    };
+
+    let api_payload = ApiPayload {
+        service: "NETWORK_TOKEN".to_string(),
+        card_data: jwt,
+        order_data,
+        key_id,
+        should_send_token: true,
+        sub_merchant_id: "visa_sbx_working".to_string(),
+    };
+
+    let mut request = services::Request::new(
+        services::Method::Post,
+        tokenization_service.generate_token_url.as_str(),
+    );
+    request.add_header(headers::CONTENT_TYPE, "application/json".into());
+    request.add_header(
+        headers::AUTHORIZATION,
+        tokenization_service
+            .token_service_api_key
+            .peek()
+            .clone()
+            .into_masked(),
+    );
+    request.set_body(RequestContent::Json(Box::new(api_payload)));
+
+    let response = services::call_connector_api(state, request, "generate_token")
+        .await
+        .change_context(errors::NetworkTokenizationError::ResponseDeserializationFailed);
+
+    let res = response
+        .map_err(|error| {
+            error.change_context(errors::NetworkTokenizationError::ResponseDeserializationFailed)
+        })
```

Comment:
We can directly call `change_context()`? Please make the same change elsewhere in this file where you've called `map_err(|error| error.change_context(...))`.